### PR TITLE
Add a missing property and fix a type for `prosemirror-*` module

### PR DESCRIPTION
**Pull request by percy507**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- Add `disableDropCursor ` to NodeSpec. 
  - https://github.com/ProseMirror/prosemirror-dropcursor/blob/master/src/dropcursor.js#L9
- Remove boolean type of getPos
  - https://prosemirror.net/docs/ref/#view.EditorProps.nodeViews

### CODE REVIEW

File: types/prosemirror-model/index.d.ts [hunk 1]
```diff
@@ -1241,6 +1242,11 @@ export interface NodeSpec {
      * to a string representation for debugging (e.g. in error messages).
      */
     toDebugString?: ((node: ProsemirrorNode) => string) | null | undefined;
+    /**
+     * @note Require `prosemirror-dropcursor` module.
+     * This property controls the showing of a drop cursor inside the node.
+     */
+    disableDropCursor?: boolean | ((view: EditorView, pos: number) => boolean);
```
Comment (Narretz) [inline types/prosemirror-model/index.d.ts:1249]: This prop is mentioned in the PM changelog, but not in the PM docs. Please open a PR with PM that adds this prop to the docs, too. Also please add the version from which disableDropCursor is supported in prosemirror-dropcursor. (1.4.0)

File: types/prosemirror-model/index.d.ts [hunk 2]
```diff
@@ -1393,7 +1399,7 @@ export interface DOMOutputSpecArray {
     8?: DOMOutputSpec | 0 | undefined;
     9?: DOMOutputSpec | 0 | undefined;
 }
-export type DOMOutputSpec = string | Node | DOMOutputSpecArray | {dom: Node, contentDOM?: Node | undefined};
+export type DOMOutputSpec = string | Node | DOMOutputSpecArray | { dom: Node; contentDOM?: Node | undefined };
```
Comment (Narretz) [inline types/prosemirror-model/index.d.ts:1402]: Looks like an unnecessary change?

File: types/prosemirror-view/index.d.ts [hunk 3]
```diff
@@ -587,7 +587,7 @@ export interface EditorProps<ThisT = unknown, S extends Schema = any> {
               [name: string]: (
                   node: ProsemirrorNode<S>,
                   view: EditorView<S>,
-                  getPos: (() => number) | boolean,
+                  getPos: () => number,
```
Comment (Narretz) [inline types/prosemirror-view/index.d.ts:590]: That's not correct: the documentation:

> For nodes, the third argument getPos is a function that can be called to get the node's current position, which can be useful when creating transactions to update it. For marks, the third argument is a boolean that indicates whether the mark's content is inline.

**Review by Narretz — CHANGES_REQUESTED** [review]
I've left comments on individual changes.

General notes:
1. I don't think we need to update prosemirror-model's version, because the spec addition of disableDropCursor doesn't really feature in prosemirror-model, it's only functional in the dropcursor module.
2. Just a thought: do we now have cyclical dependency between prosemirror-view and prosemirror-model, now that the latter imports EditorView? Does typescript even care?

**Comment by percy507** [comment]
> I've left comments on individual changes.
> 
> General notes:
> 
> 1. I don't think we need to update prosemirror-model's version, because the spec addition of disableDropCursor doesn't really feature in prosemirror-model, it's only functional in the dropcursor module.
> 2. Just a thought: do we now have cyclical dependency between prosemirror-view and prosemirror-model, now that the latter imports EditorView? Does typescript even care?

Ah, I think you are right.
I will close this PR.
